storage/flush.go

```diff
@@ -89,8 +86,24 @@ func (m *flushManager) Flush(curr time.Time) error {
 
 	multiErr := xerrors.NewMultiError()
 	for _, ns := range namespaces {
+		// Flush first because we will only snapshot if there are no outstanding flushes
 		flushTimes := m.namespaceFlushTimes(ns, curr)
 		multiErr = multiErr.Add(m.flushNamespaceWithTimes(ns, flushTimes, flush))
+
+		var (
+			blockSize          = ns.Options().RetentionOptions().BlockSize()
+			snapshotBlockStart = m.snapshotBlockStart(ns, curr)
+			prevBlockStart     = snapshotBlockStart.Add(-blockSize)
+		)
+
+		// Don't perform any snapshots until the previous blocks flush has completed
```

Comment:
what if the namespace has already written a fileset file for that time, do we even need to snapshot at all in that case?

Reply:
Is that possible? We pass the same "current time" to both methods, so either we've reached a point where we should be flushing one block (and snapshotting the other), or we shouldn't be flushing the current block but should be snapshotting.

Does that make sense?

Reply:
Yeah i didn't follow that you're checking if flush succeeded for blockstart `t-1` before snapshotting for blockstart `t`. Could you reword the comment to make that clearer.